pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/assignment/segment/SegmentAssignment.java

```diff
@@ -23,37 +23,42 @@
 import org.apache.commons.configuration.Configuration;
 import org.apache.helix.HelixManager;
 import org.apache.pinot.common.config.TableConfig;
+import org.apache.pinot.common.utils.InstancePartitionsType;
+import org.apache.pinot.controller.helix.core.assignment.InstancePartitions;
 
 
 /**
- * Strategy to assign segment to instances or rebalance all segments in a table.
+ * Interface for segment assignment and table rebalancing.
  */
-public interface SegmentAssignmentStrategy {
+public interface SegmentAssignment {
```

Comment:
Can you add the `TODO` document for adding support for custom segment assignment strategy?
I agree that it's not the right time to add the interface since we need the proper design for it. I can see multiple ways to achieve custom segment assignment (e.g. cost model - currently the cost is based on # of segments, we can extend this to add resources like storage factor into the cost).

Reply:
Good point, added.